### PR TITLE
bug/FWS-99/String_input_control

### PR DESCRIPTION
**Pull request by Eli293**
JIRA: link to jira ticket
https://champlainsaintlambert.atlassian.net/browse/FWS-99
## Context:
This ticket is about the addition of input validation for the booking.
 
## Changes

- Added Input validation
- This affects the booking feature
- This affects Members

### CODE REVIEW

File: fitwsarah_frontend/src/views/AppointmentPage/Calendar.js [hunk 1]
```diff
@@ -108,7 +112,8 @@ const AvailabilitiesCalendar = ({onChange}) => {
         <div className='time-picker'>
             <div>
             <div class="focusable time-picker" tabindex="0" role="group">
-            {!noAvailabilities &&
+              {loading && <svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 24 24"><path fill="currentColor" d="M12,1A11,11,0,1,0,23,12,11,11,0,0,0,12,1Zm0,19a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z" opacity="0.25"/><path fill="currentColor" d="M12,4a8,8,0,0,1,7.89,6.7A1.53,1.53,0,0,0,21.38,12h0a1.5,1.5,0,0,0,1.48-1.75,11,11,0,0,0-21.72,0A1.5,1.5,0,0,0,2.62,12h0a1.53,1.53,0,0,0,1.49-1.3A8,8,0,0,1,12,4Z"><animateTransform attributeName="transform" dur="0.75s" repeatCount="indefinite" type="rotate" values="0 12 12;360 12 12"/></path></svg>}
+                          {!loading && !noAvailabilities &&
```
Comment (emilegirardGit) [inline fitwsarah_frontend/src/views/AppointmentPage/Calendar.js:116]: Can you explain what this does?

Comment (Eli293) [reply fitwsarah_frontend/src/views/AppointmentPage/Calendar.js:116]: This sets a loading animation whenever the server fetches all the availabilities, to make it clearer to the user.

**Review by Abdur-Siddiqui — APPROVED** [review]
looks good

**Review by AlexeiTimbro — APPROVED** [review]
Looks good

**Pull request merged**